packages/smooth_app/lib/pages/page_manager.dart

```diff
@@ -84,10 +84,14 @@ class PageManagerState extends State<PageManager> {
           onTap: (int index) {
             if (_currentPage == BottomNavigationTab.Scan &&
                 _pageKeys[index] == BottomNavigationTab.Scan) {
-              InheritedDataManager.of(context).resetShowSearchCard(true);
+              if (!InheritedDataManager.of(context).showSearchCard) {
```

Comment:
Please use a final variable for `InheritedDataManager.of(context)`, that is used 4 times.